src/test/kotlin/cacophony/semantic/names/NameResolverTest.kt

```diff
@@ -1214,5 +1214,41 @@ class NameResolverTest {
             verify(exactly = 1) { diagnostics.report(NRDiagnostics.IllegalFunctionalArgument("x"), functionalArgumentRange) }
             confirmVerified(diagnostics)
         }
+
+        @Test
+        fun `error when finds duplicated argument names`() {
+            // let f = [x: Int, x: Bool] -> Int => 0
+
+            // given
+            val firstArgumentRange = Location(1) to Location(1)
+            val secondArgumentRange = Location(2) to Location(2)
+            val ast =
+                block(
+                    functionDeclaration(
```

Comment:
nit, but if body is `lit(0)`, maybe let's make this a intFunctionDeclaration

Reply:
good catch